From 481d409f896326b19630f6975f9559f9b62f52b5 Mon Sep 17 00:00:00 2001
From: "Xie, Jiabao(Jimbo)" <Jiabao.Xie@amd.com>
Date: Tue, 15 Jul 2025 22:32:58 +0000
Subject: [PATCH 3/3] SWDEV-460151 - add gfx1201 to amd-staging clr (#183)

* SWDEV-460151 - add gfx1201 to amd-staging clr

* SWDEV-460151 - removed pal macro

---------

Co-authored-by: Jimbo Xie <jiabaxie@amd.com>
---
 rocclr/device/pal/paldevice.cpp   | 1 +
 rocclr/device/pal/palsettings.cpp | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/rocclr/device/pal/paldevice.cpp b/rocclr/device/pal/paldevice.cpp
index 03a087b0f..59edc1a23 100644
--- a/rocclr/device/pal/paldevice.cpp
+++ b/rocclr/device/pal/paldevice.cpp
@@ -102,6 +102,7 @@ static constexpr PalDevice supportedPalDevices[] = {
   {11, 0,  3,  Pal::GfxIpLevel::GfxIp11_0, "gfx1103",       Pal::AsicRevision::HawkPoint2},
   {11, 5,  0,  Pal::GfxIpLevel::GfxIp11_5, "gfx1150",       Pal::AsicRevision::Strix1},
   {11, 5,  1,  Pal::GfxIpLevel::GfxIp11_5, "gfx1151",       Pal::AsicRevision::StrixHalo},
+  {12, 0,  1,  Pal::GfxIpLevel::GfxIp12,   "gfx1201",       Pal::AsicRevision::Navi48},
 };
 
 static std::tuple<const amd::Isa*, const char*> findIsa(uint32_t gfxipMajor, uint32_t gfxipMinor,
diff --git a/rocclr/device/pal/palsettings.cpp b/rocclr/device/pal/palsettings.cpp
index 8361a950c..0cd5ca70a 100644
--- a/rocclr/device/pal/palsettings.cpp
+++ b/rocclr/device/pal/palsettings.cpp
@@ -161,6 +161,8 @@ bool Settings::create(const Pal::DeviceProperties& palProp,
   amd::Os::getAppPathAndFileName(appName, appPathAndName);
 
   switch (palProp.revision) {
+    // Fall through for Navi4x ...
+    case Pal::AsicRevision::Navi48:
     // Fall through for Navi3x ...
     case Pal::AsicRevision::Navi33:
     case Pal::AsicRevision::Navi32:
-- 
2.43.0

